src/test/java/uk/gov/companieshouse/itemhandler/kafka/OrdersKafkaConsumerTest.java

```diff
@@ -25,96 +31,161 @@
+        ordersKafkaConsumer.handleMessage(createTestMessage(ORDER_RECEIVED_TOPIC));
+        // Then
+        verify(ordersKafkaConsumer, times(1)).republishMessageToTopic(anyString(), anyString(), anyString());
+    }
+
+    @Test
+    public void republishMessageNotCalledForFirstRetryMessageOnRetryableErrorException() {
+        // Given & When
+        doThrow(new RetryableErrorException(PROCESSING_ERROR_MESSAGE)).when(ordersKafkaConsumer).logMessageReceived(any());
+        ordersKafkaConsumer.handleMessage(createTestMessage(ORDER_RECEIVED_TOPIC_RETRY));
         // Then
-        verify(chKafkaConsumerGroupRetry, times(3)).retry(anyInt(), any());
+        verify(ordersKafkaConsumer, times(0)).republishMessageToTopic(anyString(), anyString(), anyString());
     }
 
     @Test
-    public void mainListenerExceptionIsCorrectlyHandled() throws InterruptedException, ExecutionException, SerializationException {
+    public void republishMessageNotCalledForFirstErrorMessageOnRetryableErrorException() {
         // Given & When
-        doThrow(new OrderProcessingException()).when(ordersKafkaConsumer).processOrderReceived(ORDER_RECEIVED_URI);
-        OrderProcessingException exception = Assertions.assertThrows(OrderProcessingException.class, () -> {
-            ordersKafkaConsumer.processOrderReceived(ORDER_RECEIVED_URI);
+        doThrow(new RetryableErrorException(PROCESSING_ERROR_MESSAGE)).when(ordersKafkaConsumer).logMessageReceived(any());
+        ordersKafkaConsumer.handleMessage(createTestMessage(ORDER_RECEIVED_TOPIC_ERROR));
+        // Then
+        verify(ordersKafkaConsumer, times(0)).republishMessageToTopic(anyString(), anyString(), anyString());
+    }
+
+    @Test
+    public void republishMessageNotCalledOnNonRetryableErrorException() {
+        // Given & When
+        doThrow(new OrderProcessingException()).when(ordersKafkaConsumer).logMessageReceived(any());
+        ordersKafkaConsumer.handleMessage(createTestMessage(ORDER_RECEIVED_TOPIC));
+        // Then
+        verify(ordersKafkaConsumer, times(0)).republishMessageToTopic(anyString(), anyString(), anyString());
+    }
+
+    @Test
+    public void mainListenerExceptionIsCorrectlyHandled() {
+        // Given & When
+        doThrow(new RetryableErrorException(PROCESSING_ERROR_MESSAGE)).when(ordersKafkaConsumer).processOrderReceived(any());
+        RetryableErrorException exception = Assertions.assertThrows(RetryableErrorException.class, () -> {
+            ordersKafkaConsumer.processOrderReceived(any());
         });
         // Then
-        String expectedMessage = "Order processing failed.";
         String actualMessage = exception.getMessage();
-        Assert.assertThat(actualMessage, Matchers.is(expectedMessage));
-        verify(ordersKafkaConsumer, times(1)).processOrderReceived(anyString());
+        Assert.assertThat(actualMessage, Matchers.is(PROCESSING_ERROR_MESSAGE));
+        verify(ordersKafkaConsumer, times(1)).processOrderReceived(any());
```

Comment:
Sorry, why is this not `processOrderReceivedRetry(...)`?

Reply:
This test has been removed.

Reply:
Since this is just a unit test, it doesn't simulate the message being propagated to the next topic. There are integration tests that cover those scenarios.